ctags/json.go

```diff
@@ -51,13 +49,9 @@ type lockedParser struct {
 const parseTimeout = time.Minute
 
 // Parse wraps go-ctags Parse. It lazily starts the process and adds a timeout
-// around parse requests. Additionally it serializes access to the parsing
-// process. The timeout is important since we occasionally come across
-// documents which hang universal-ctags.
+// around parse requests. The timeout is important since we occasionally come
+// across documents which hang universal-ctags.
 func (lp *lockedParser) Parse(name string, content []byte) ([]*Entry, error) {
-	lp.mu.Lock()
-	defer lp.mu.Unlock()
```

Comment:
So this is fine to do now since we create and call the parser for each shard? Is that obvious from the code path that calls Parse? Is it worth documenting this (maybe is documented in a later commit)

Reply:
Yes on the struct comment I wrote that it's only safe for single-threaded use. Let me know if there's a more obvious way to document this!